invenio_app_ils/permissions.py

```diff
@@ -89,6 +97,73 @@ def __init__(self, record):
         )
 
 
+class LoanExtendPermission(Permission):
+    """Return Permission to evaluate if the user can extend the loan."""
+
+    def __init__(self, record):
+        """Constructor."""
+        self.record = record
+
+        # NOTE: extra validation if the user is not librarian or admin
+        if current_user.id == int(record["patron_pid"]):
+            self.validate()
+
+        super().__init__(
+            UserNeed(int(record["patron_pid"])), backoffice_access_action
+        )
+
+    def validate(self):
+        """Validate if the loan can be extended."""
+        if not self.validate_extend_enabled():
+            raise InvalidLoanExtendError(
+                "You can extend this loan {} days before it "
+                "expires!".format(
+                    current_app.config["ILS_LOAN_WILL_EXPIRE_DAYS"])
+            )
+
+        if not self.validate_max_extensions():
+            raise InvalidLoanExtendError(
+                "You have reached the max number of extensions for THIS loan!"
+            )
+
+        if not self.validate_pending_loans():
+            raise InvalidLoanExtendError(
+                "There is a high demand for this literature!"
+            )
+
+        if self.validate_is_overdue():
+            raise InvalidLoanExtendError(
+                "This loan is overdue, its end date has passed!"
+            )
+
+    def validate_extend_enabled(self):
+        """Validate loan is in the period that can be extended."""
+        end_date = arrow.get(self.record["end_date"])
+        return (
+            (arrow.get().utcnow() - end_date).days <=
+            current_app.config["ILS_LOAN_WILL_EXPIRE_DAYS"]
+        )
+
+    def validate_max_extensions(self):
+        """Validate if we have reached the max extension count."""
+        return (
+            self.record["extension_count"] <=
+            circulation_default_extension_max_count(self.record)
+        )
+
+    def validate_pending_loans(self):
+        """Validate the loaned document has no pending loan requests."""
```

Comment:
here we should check not if we have requests, but if the document is  overbooked. Because if we have 4 items but 2 requests I don't see the reason why not to extend.